pkg/semtech-loramac/contrib/semtech_loramac.c

```diff
@@ -63,30 +64,21 @@
 
 #define SEMTECH_LORAMAC_MSG_QUEUE                   (16U)
 #define SEMTECH_LORAMAC_LORAMAC_STACKSIZE           (THREAD_STACKSIZE_DEFAULT)
+static msg_t _semtech_loramac_msg_queue[SEMTECH_LORAMAC_MSG_QUEUE];
 static char _semtech_loramac_stack[SEMTECH_LORAMAC_LORAMAC_STACKSIZE];
 kernel_pid_t semtech_loramac_pid;
-kernel_pid_t semtech_loramac_handler_pid;
 
+sx127x_t sx127x;
```

Comment:
Do you think it could make sense to also isolate these kind of variables in a struct?

Reply:
(e.g this could come from the netdev_t pointer)

Reply:
The problem is that it is used by the [radio adaption code](https://github.com/RIOT-OS/RIOT/blob/master/pkg/semtech-loramac/contrib/semtech_loramac_radio.c#L36) as an `extern` variable. Using a get/set function for passing/retrieving the pointer as you initially did won't be better I think.
The best solution would be to pass it the all semtech-loramac function but then it will require massive patches in the upstream code.